### PR TITLE
New release updates to drop pypy support

### PR DESCRIPTION
**Pull request by kmpaul**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

With Dask dropping pypy support, I would like to remove pypy migrations.  I also need to update the entry points.

### CODE REVIEW

**Comment by kmpaul** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,11 +10,11 @@ source:
   sha256: 0a04f1d7d35a06cdff506593330d4414ea242c9172498ce191f5742eac499e17
 
 build:
-  skip: true  # [win]
-  number: 0
+  skip: true  # [win or python_impl == 'pypy']
```
Comment (jakirkham) [inline recipe/meta.yaml:13]: Curious why Windows is skipped? Are we missing a dependency?

Comment (kmpaul) [reply recipe/meta.yaml:13]: That's historical, and it had more to do with MPI dependence, not Dask-MPI. I can look into it more and see if we can build in windows.

Comment (jakirkham) [reply recipe/meta.yaml:13]: Ok it seems like we have `mpi4py`, which would be the biggest blocker I would think.

Comment (kmpaul) [reply recipe/meta.yaml:13]: Trying it out with #23.

**Pull request merged**